[Bug]: error when trying to run command or clone

### osx-proxmox-next version

either installer bash commands

### Proxmox VE version

9.2.5

### macOS version

Ventura 13

### CPU type

Intel

### Which stage failed?

plan (generating config)

### What happened?

Try to run either script with or without dependencies causes an error and cannot be executed on the latest Proxmox release 9.2.5 
Please review this, thank you. 

Loading...
pushd: error retrieving current directory: getcwd: cannot access parent directories: No such file or directory

[ERROR] in line 792: exit code 1: while executing command pushd "$TEMP_DIR" > /dev/null

root@pve:~/osx-proxmox-next# 

### Relevant logs

```shell

```

--- a/install.sh
+++ b/install.sh
@@ -72,10 +72,15 @@ launch() {
 }
 
 nuke_stale() {
-  # Remove previous install to prevent stale code/venv from persisting
+  # Clear previous install to prevent stale code/venv from persisting.
+  # Empty the directory rather than rm -rf'ing it: if the invoking shell's
+  # cwd is $REPO_DIR (e.g. user cloned it manually and cd'd in), removing
+  # the directory itself leaves that shell's getcwd() unresolvable, which
+  # breaks any later command needing it (pushd, pwd -P, etc.) even after
+  # this script exits.
   if [[ -d "$REPO_DIR" ]]; then
     log "Removing previous install..."
-    rm -rf "$REPO_DIR"
+    find "$REPO_DIR" -mindepth 1 -delete
   fi
 }
 

--- a/scripts/bash/osx-proxmox-next.sh
+++ b/scripts/bash/osx-proxmox-next.sh
@@ -788,6 +788,12 @@ with open(path, 'wb') as f:
   msg_ok "Built OpenCore boot disk"
 }
 
+# pushd needs a resolvable cwd to save onto the dir stack; if the caller's
+# shell is sitting in a directory that got removed from under it (e.g. by a
+# prior install script deleting its own repo checkout), getcwd() fails and
+# pushd aborts. Land somewhere known-good first so that can't happen here.
+cd "${HOME:-/root}" 2>/dev/null || cd /
+
 TEMP_DIR=$(mktemp -d)
 pushd "$TEMP_DIR" >/dev/null
 

